### PR TITLE
add support for Debian/Ubuntu

### PR DESCRIPTION
**Pull request by thobianchi**
This simple PR will add support for Debian and Ubuntu.

Differently from Red Hat, Debian will create a file /var/run/reboot-required if is required ( kernel update ) a reboot.

### CODE REVIEW

**Comment by dschier-wtd** [comment]
Hey, 
thanks a lot for your contribution. I really appreciate every help :)

As far as I know the following code does not work on debianoid systems, but will install a ton of packages:

```
- package:
    name: "*"
    state: "latest"
```
Furthermore, I am unsure, if one has to update the package cache manually for debian systems :/

Could you also have a look at the tests (molecule/travis) and provide a proper configuration to test debian and ubuntu in your supported version? Additionally, I request the addition to the meta/main.yml for your desired OS Versions.

**Comment by thobianchi** [comment]
from official [documentation](https://docs.ansible.com/ansible/latest/modules/apt_module.html) of *apt* module:

> update_cache: Run the equivalent of apt-get update before the operation.

Directly in the examples there is:

```
- name: Upgrade all packages to the latest version
  apt:
    name: "*"
    state: latest
```
Its intention is to upgrade all packages not install every package in the repo, I tested it.

Reguarding travis build I'm not an expert, I will try to adapt to Debian and Ubuntu, for now I tested it against a Debian 10.

**Comment by thobianchi** [comment]
I'm getting this error:
```
TASK [Create molecule instance(s)] *********************************************
    changed: [localhost] => (item=debian10)
    
    TASK [Wait for instance(s) creation to complete] *******************************
    FAILED - RETRYING: Wait for instance(s) creation to complete (300 retries left).
    failed: [localhost] (item=None) => {"attempts": 2, "censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result", "changed": false}
    fatal: [localhost]: FAILED! => {"censored": "the output has been hidden due to the fact that 'no_log: true' was specified for this result", "changed": false}
```
but I can't figure out what's wrong, any ideas?

**Comment by dschier-wtd** [comment]
You can run molecule with --debug to have a better understanding of whats going wrong. I think, this may be a problem with debian/ubuntu images not supporting init per default.

I'll have a look and will update you.

File: molecule/default/molecule.yml [hunk 1]
```diff
@@ -27,29 +27,25 @@ driver:
...
     volumes:
       - /sys/fs/cgroup:/sys/fs/cgroup:ro
     tmpfs:
       - /run
       - /tmp
-    # privileged: true
-    groups:
-      - ${pygroup:-python}
+    privileged: true
+    pre_build_image: true
```
Comment (thobianchi) [inline molecule/default/molecule.yml:38]: Changed a bit this file, I followed this [article](https://www.jeffgeerling.com/blog/2018/testing-your-ansible-roles-molecule), as you can see I deleted the `dockerfile.j2` to adopt these pre-built image with **ansible** support and systemd and are working very well with zero effort.

File: tasks/Fedora.yml [hunk 2]
```diff
@@ -28,12 +28,13 @@
     - upgrade
 
 - name: Check System Reboot
-  command: tracer
+  command: needs-restarting -r
```
Comment (thobianchi) [inline tasks/Fedora.yml:31]: tracer method worked when I tested locally but on travis build hangs, I switched to `dnf-utils` and now is working as expected like centos

**Comment by thobianchi** [comment]
solved issues on tests, it would be nice to add more tests, I will think about it and maybe open a new PR if it pleases you @daniel-wtd

File: .travis.yml [hunk 3]
```diff
@@ -13,7 +13,7 @@ services:
 
 install:
   # Install test dependencies.
-  - pip install ansible==2.7.10 molecule[docker]
+  - pip install ansible==2.9.2 molecule[docker]
```
Comment (daniel-schier) [inline .travis.yml:16]: You can switch to "pip install ansible molecule[docker]". The version was only needed due to some issues in 2.8.x, which were fixed in 2.9.1

File: tasks/Fedora.yml (hunk 2 above)
Comment (daniel-schier) [reply tasks/Fedora.yml:31]: looks awesome, thanks.

**Review by dschier-wtd — CHANGES_REQUESTED** [review]
Hi,
thanks again for your huge effort here. I really appreciate, if somebody is supporting and keeping up with the work.

Unfortunately, removing the dockerfile is not a good idea, because your removed 2 major features:

- playbooks are no longer running as user "ansible" in the container
- autobuild images from upstream are not maintained from jeff / geerlinguy (nevertheless, he is a cool guy and doing a lot of cool work)

You also introduced a new issue:

- running a privileged image for a simple task like updating is not recommended

I suggest:

- add the dockerfile again
- apply the needed systemd changes to the official upstream images in this dockerfile

**Pull request closed without merging**